src/store/student-store.js

```diff
@@ -81,18 +80,31 @@ function removeItemFromAllStudents(itemAddress) {
         }
     }
 }
-function addItemsToStudent(studentId, equipmentAddresses){
-    let student = store.getStudentById(studentId);
-    equipmentAddresses.forEach(address => {
-        let result = readAddress(address);
-        if(result.type == 'item'){
+
+function addItemsToStudent(studentId, equipment, dueDateTime){
+    let student = students[studentId];
+    equipment.forEach(equip => {
+        let address = equip.address;
+        let type = readAddress(address).type;
+        if (type === 'item') {
             student.items.push(ItemStore.getItemByAddress(address));
-        }
-        else if (result.type == 'model') {
-            student.models.push(ModelStore.getModelByAddress(address));
+        } else if (type === 'model') {
+            let index = student.models.findIndex(model => model.address === address);
```

Comment:
Nit: why not just `let model = student.models.find(model => model.address === address);`?